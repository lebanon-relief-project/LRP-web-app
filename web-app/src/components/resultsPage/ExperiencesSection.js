import React from "react";
import styled from "styled-components";
import colours from "../../styles/Colours";
import devices from "../../styles/Devices";
import ResultsPageLeftBackground from "../../assets/images/background/ResultsPageLeftBackground.svg";
import ResultsPageRightBackground from "../../assets/images/background/ResultsPageRightBackground.svg";
import ResultsPagePeopleIcons from "../../assets/images/background/ResultsPagePeopleIcons.svg";
import ResultsPageStatistics from "../../assets/images/background/ResultsPageStatistics.svg";

const ExperiencesSection = () => {
  return (
    <StyledSection>
      <HeaderWrapper>
        <HeaderLeftImageWrapper>
          <img src={ResultsPageLeftBackground} alt={"background"} />
        </HeaderLeftImageWrapper>

        <HeaderRightImageWrapper>
          <img src={ResultsPageRightBackground} alt={"background"} />
        </HeaderRightImageWrapper>
      </HeaderWrapper>

      <CentralWrapper>
        <CentralContentWrapper>
          <StyledHeader>From the experiences you've shared...</StyledHeader>
          <TextWrapper>
            <StyledParagraph>
              We have pulled together some tailored advice and exercises to help
              you cope with trauma and how you are feeling.
            </StyledParagraph>
            <StyledParagraph>
              <img src={ResultsPagePeopleIcons} alt={"76%"} />
            </StyledParagraph>
            <StyledParagraph>
              <img src={ResultsPageStatistics} alt={"people"} />
            </StyledParagraph>
          </TextWrapper>
        </CentralContentWrapper>
      </CentralWrapper>
    </StyledSection>
  );
};

const TextWrapper = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
`;

const CentralWrapper = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  z-index: 2;
  background-color: transparent;
  justify-content: center;
  position: absolute;
  align-items: flex-start;
  height: 363px;
  width: 100%;

  @media (max-width: ${devices.mobile}) {
    margin-top: 20px;
    height: 371px;
  }
`;

const CentralContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background-color: transparent;
  margin-left: 16.7%;
  max-width: 765px;

  @media (max-width: ${devices.mobile}) {
    margin-left: 18px;
  }
`;

const HeaderWrapper = styled.div`
  display: flex;
  flex: 1;
  z-index: 1;
  justify-content: space-between;
`;

const HeaderLeftImageWrapper = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-start;
  align-items: center;

  @media (max-width: ${devices.ipadpro}) {
    display: none;
  }
`;

const HeaderRightImageWrapper = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-items: center;

  @media (max-width: ${devices.ipadpro}) {
    display: none;
  }
`;

const StyledSection = styled.div`
  display: flex;
  background: ${colours.darkBlue};
  margin: 0;
  margin-top: 64px;
  padding: 0;
  width: 100%;
  height: 363px;
  z-index: 0;

  @media (max-width: ${devices.mobile}) {
    height: 416px;
  }
`;

const StyledHeader = styled.h2`
  font-weight: bold;
  font-size: 38px;
  margin: 0;
  width: inherit;
  color: ${colours.white};
`;

const StyledParagraph = styled.p`
  margin: 0;
  margin-bottom: 1rem;
  font-size: 16px;
  color: ${colours.white};
`;

export default ExperiencesSection;
