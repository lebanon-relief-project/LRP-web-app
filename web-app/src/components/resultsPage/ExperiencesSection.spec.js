import React from "react";
import { render, wait } from "@testing-library/react";
import ExperiencesSection from "./ExperiencesSection";
import "jest-styled-components";
import "@testing-library/jest-dom/extend-expect";

jest.mock("../../services/results.service", () => {
  return {
    getPercentage: jest.fn().mockResolvedValue({
      default_percentage: 0.76,
    }),
  };
});

describe("the TraumaticEvents component", () => {
  let container;

  it("should match the snapshot", async () => {
    ({ container } = render(<ExperiencesSection />));

    await wait();

    expect(container).toMatchSnapshot();
  });
});
