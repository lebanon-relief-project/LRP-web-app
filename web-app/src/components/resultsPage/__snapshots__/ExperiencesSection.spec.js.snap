// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the TraumaticEvents component should match the snapshot 1`] = `
.c9 {
  margin-right: 20px;
}

.c11 {
  font-family: Raleway;
  font-size: 60px;
  font-style: normal;
  font-weight: 800;
  line-height: 46px;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  text-align: left;
  margin-right: 2%;
}

.c12 {
  width: 60%;
  font-family: Raleway;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  text-align: left;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 2;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: 363px;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: transparent;
  margin-left: 16.7%;
  max-width: 765px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #002766;
  margin: 0;
  margin-top: 64px;
  padding: 0;
  width: 100%;
  height: 363px;
  z-index: 0;
}

.c6 {
  font-weight: bold;
  font-size: 38px;
  margin: 0;
  width: inherit;
  color: #ffffff;
}

.c8 {
  margin: 0;
  margin-bottom: 1rem;
  font-size: 16px;
  color: #ffffff;
}

@media (max-width:576px) {
  .c4 {
    margin-top: 20px;
    height: 371px;
  }
}

@media (max-width:576px) {
  .c5 {
    margin-left: 18px;
  }
}

@media (max-width:1025px) {
  .c2 {
    display: none;
  }
}

@media (max-width:1025px) {
  .c3 {
    display: none;
  }
}

@media (max-width:576px) {
  .c0 {
    height: 416px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="background"
          src="ResultsPageLeftBackground.svg"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="background"
          src="ResultsPageRightBackground.svg"
        />
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <h2
          class="c6"
        >
          From the experiences you've shared...
        </h2>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            We have pulled together some tailored advice and exercises to help you cope with trauma and how you are feeling.
          </div>
          <div
            class="c8"
          >
            <img
              alt="Filled Person"
              class="c9"
              data-testid="resultsPercentage_image_0"
              src="FilledPerson.svg"
            />
            <img
              alt="Filled Person"
              class="c9"
              data-testid="resultsPercentage_image_1"
              src="FilledPerson.svg"
            />
            <img
              alt="Filled Person"
              class="c9"
              data-testid="resultsPercentage_image_2"
              src="FilledPerson.svg"
            />
            <img
              alt="Filled Person"
              class="c9"
              data-testid="resultsPercentage_image_3"
              src="FilledPerson.svg"
            />
            <img
              alt="Filled Person"
              class="c9"
              data-testid="resultsPercentage_image_4"
              src="FilledPerson.svg"
            />
            <img
              alt="Filled Person"
              class="c9"
              data-testid="resultsPercentage_image_5"
              src="FilledPerson.svg"
            />
            <img
              alt="Filled Person"
              class="c9"
              data-testid="resultsPercentage_image_6"
              src="FilledPerson.svg"
            />
            <img
              alt="Filled Person"
              class="c9"
              data-testid="resultsPercentage_image_7"
              src="FilledPerson.svg"
            />
            <img
              alt="Empty Person"
              class="c9"
              data-testid="resultsPercentage_image_8"
              src="EmptyPerson.svg"
            />
            <img
              alt="Empty Person"
              class="c9"
              data-testid="resultsPercentage_image_9"
              src="EmptyPerson.svg"
            />
          </div>
          <div
            class="c8"
          >
            <div
              class="c10"
            >
              <div
                class="c11"
              >
                76
                %
              </div>
              <div
                class="c12"
              >
                of people affected by this experience feel like ictum nullam ligula lorem ut vitae. Purus blandit imperdiet nibh amet.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
